modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java

```diff
@@ -216,19 +232,73 @@ public boolean removeClassLoader(ArtifactClassLoader artifactClassLoader) {
 
   @Override
   public final URL findResource(final String name) {
-    URL resource = null;
     String normalizedName = normalize(name, true);
+    // Check exported resources and all matching artifacts
     final List<ArtifactClassLoader> artifactClassLoaders = resourceMapping.get(normalizedName);
     if (artifactClassLoaders != null) {
       for (ArtifactClassLoader artifactClassLoader : artifactClassLoaders) {
-        resource = artifactClassLoader.findResource(normalizedName);
-        if (resource != null) {
-          break;
+        URL url = artifactClassLoader.findResource(normalizedName);
+        if (url != null) {
+          return url;
+        }
+      }
+    } else if (name.startsWith(RESOURCE_PREFIX)) {
+      Matcher matcher = GAV_PATTERN.matcher(name);
+      // Check for specific artifact requests
+      if (matcher.matches()) {
+        String groupId = matcher.group(1);
+        String artifactId = matcher.group(2);
+        String version = matcher.group(3);
+        String resource = matcher.group(4);
+        LOGGER.debug("Region request for '{}' in group '{}', artifact '{}' and version '{}'.", resource, groupId, artifactId,
+                     version);
+        String normalizedResource = normalize(resource, true);
+
+        // Check whether it's an exported resource from a matching artifact
+        List<ArtifactClassLoader> exportingArtifactClassLoaders = resourceMapping.get(normalizedResource);
+        if (exportingArtifactClassLoaders != null) {
+          for (ArtifactClassLoader artifactClassLoader : exportingArtifactClassLoaders) {
+            BundleDescriptor descriptor = artifactClassLoader.getArtifactDescriptor().getBundleDescriptor();
+            if (isRequestedArtifact(descriptor, groupId, artifactId, version, () -> {
+              LOGGER.warn("Required version '{}' for artifact '{}:{}' not found. Searching in available version '{}'...",
+                          version, descriptor.getGroupId(), descriptor.getArtifactId(), descriptor.getVersion());
+              return true;
```

Comment:
Shouldn't return false on this case?

Reply:
My thinking here is that when requesting a plugin resource, the requested version might not be what ends up in the app and there will only be one plugin version, so if there's a partial match we should try and search in it. Since the resource is part of the API (being exported) chances are you'll find what you need.

Reply:
You are right should try if it matches the groupId:artifactId as the resolved plugin (in case of plugins) could be affected by semver so you don't know exactly which version to reference.